src/internal/components/cartesian-chart/left-labels.tsx

```diff
@@ -42,15 +45,15 @@ function LeftLabels({
 
   const yOffset = axis === 'x' && scale.isCategorical() ? Math.max(0, scale.d3Scale.bandwidth() - 1) / 2 : 0;
 
-  const cacheRef = useRef<{ [label: string]: number }>({});
+  const labelToBoxCache = useRef<{ [label: string]: DOMRect }>({});
```

Comment:
Maybe this logic could be somehow contained inside `formatTicks`? We could pass it `virtualTextRef.current` instead of `getLabelSpace` and this component would be a bit cleaner

Reply:
then we'd also need to pass down the cache and repeat the same for the bottom labels

Reply:
The box information (or just the width if that's the only part we use) of each line could come with each item returned from `getVisibleTicks`?

Reply:
It could, but feels like a bigger refactoring that is better to be done separately. I would probably apply the formatting first with line-splitting and then in a separate function measure and cache the widths and heights.